docs/design.md

```diff
@@ -1,45 +1,29 @@
+* [Metadata Discovery](#metadata-discovery): Discovers possible metadata sources and enriches the profiles collect from the host system Parca Agent runs on.
+* [Send data to server](#send-data-to-server): Upload available symbol data and [pprof](https://github.com/google/pprof)-formatted profile to the remote server.
 
 ![Parca Agent Architecture Diagram](https://docs.google.com/drawings/d/18xXj1Tjt9l-iuR3gse1lqI4QA2XTCQOylC5kc2cVMT4/export/svg)
 
-## Target Discovery
-
-The Parca Agent works by attaching the profiler to a [Linux cgroups](https://en.wikipedia.org/wiki/Cgroups), so the cgroups to attach to must first be discovered through some mechanism. Parca Agent currently has two primary mechanisms to discover cgroups: Kubernetes and systemd. Besides the cgroups, the target discovery also provides the labels to label a series of profiles being sent to the server.
-
-### Kubernetes
-
-To discover cgroups to profile in Kubernetes, Parca Agent first discovers all Pods running on the node it is on, then discovers the primary PID of the cgroup using Kubernetes CRI (container runtime interface). For profiling purposes a `perf_event` cgroup is required, which is read from `/proc/PID/cgroup`.
-
-> Note: Unfortunately, Kubernetes CRI does not actually have the retrieving PIDs formally specified, because CRI is intended for any type of sandboxing, including virtual machines (which wouldn't have a PID in the same way). While there are some recommendations, in practice retrieving the PID actually requires a direct integration with each container runtime, because runtimes are inconsistent and don't appear to follow the recommendation. Parca Agent has explicit support for docker, containerd, and cri-o.
-
-Labels provided by the Kubernetes discovery are: `node`, `namespace`, `pod`, `container` and `containerid`.
-
-### systemd
-
-systemd starts all services in a cgroup, and conveniently and consistently mounts the cgroup hierarchy. Unfortunately [`perf_event` is one of the few cgroups systemd does not automatically manage](https://systemd.io/CGROUP_DELEGATION/#controller-support). This means the `perf_event` cgroup needs to be separately managed by Parca Agent. It will replicate the cgroup hierarchy under `system.slice`, into `perf_event` and sync the `cgroup.procs` once per second.
-
-Labels provided by the systemd discovery are: `node` and `systemd_unit`.
-
 ## Obtaining raw data
 
-Parca Agent obtains the raw data by attaching an BPF program to a [Linux cgroup](https://en.wikipedia.org/wiki/Cgroups) using [perf_event_open](https://man7.org/linux/man-pages/man2/perf_event_open.2.html). It instructs the Kernel to call the BPF program every 100 times per second.
+Parca Agent obtains the raw data by attaching an eBPF program to a process using [perf_event_open](https://man7.org/linux/man-pages/man2/perf_event_open.2.html). It instructs the Kernel to call the BPF program every 100 times per second.
```

Comment:
I don't think this is correct, we don't attach to processes but to perf events. Perhaps we can mention that weuse the `cycles` event?

Reply:
Do you have link to put in here?